test/y2storage/devicegraph_sanitizer_test.rb

```diff
@@ -69,6 +69,29 @@
         expect(subject.errors).to be_empty
       end
     end
+
+    context "with a bcache device in the devicegraph" do
+      let(:scenario) { "bcache2.xml" }
+      let(:devicegraph) { fake_devicegraph }
+      let(:bcache_name) { "/dev/bcache1" }
+
+      context "on an architecture that supports bcache (x86_64)" do
+        let(:architecture) { :x86_64 }
+
+        it "does not contain an error" do
+          fake_scenario(scenario)
+          expect(subject.errors).to be_empty
+        end
+      end
+
+      context "on an architecture that does not support bcache (ppc)" do
+        let(:architecture) { :ppc }
+
+        it "raises an error" do
+          expect { fake_scenario(scenario) }.to raise_error(Y2Storage::Error)
```

Comment:
I don't get why an exception is raised here.

I was expecting subject.errors to contain the error about unsupported bcache, but not an exception

Reply:
Ok. I took a closer look and I found the exception is raised in the StorageManager if the user decides to abort (i.e. if the callback to inform about the error returns false).
https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/storage_manager.rb#L473

During normal execution that exception is caught. That's not the case of the `fake_scenario` helper (intentionally or not, likely nobody thought about it).

In any case, I don't think we should rely on the un-captured StorageManager exception in order to test `DevicegraphSanitizer#errors`. We should really check that `#error` contains what we want, not that our testing infrastructure explodes because we didn't provide a valid callback to report the error to the user (that's what we are testing by expecting an exception).

Reply:
OK. I prefer that approach a lot, but I didn't find out **why** I kept getting that exception. The missing callback is what I hadn't found, so I'll have another look at that.

Reply:
Now using the same technique as the other tests for that class.